python/flyft/state.py

```diff
@@ -53,12 +53,23 @@ def centers(self):
         if not hasattr(self, '_centers'):
             self._centers = np.array([self._self.center(i) for i in range(self.shape)])
         return self._centers
+    
+        
+    def volume(self, bin = None):
+        if bin is None:
+            v = self._self.total_volume()
```

Comment:
If changes above work, this would be:
```suggestion
            v = self._self.volume
```